packages/eslint-plugin/src/configs/react.js

```diff
@@ -0,0 +1,324 @@
+      },
+    },
+  },
+  env: {
+    browser: true,
+    'jest/globals': true,
+  },
+  reportUnusedDisableDirectives: true,
+  // matched relative to cwd
+  ignorePatterns: [
+    'coverage',
+    'dist',
+    'dist-storybook',
+    'etc',
+    'lib',
+    'lib-amd',
+    'lib-commonjs',
+    'node_modules',
+    'temp',
+    '**/__snapshots__',
+    '**/*.scss.ts',
+  ],
+  rules: {
+    ...configHelpers.getNamingConventionRule(false /* prefixWithI */),
+    '@typescript-eslint/no-empty-function': 'error',
+    '@typescript-eslint/no-explicit-any': 'error', // tslint: no-any
+    '@typescript-eslint/prefer-namespace-keyword': 'error',
+
+    curly: ['error', 'all'],
+    'dot-notation': 'error',
+    eqeqeq: ['error', 'always'],
+    'guard-for-in': 'error',
+    'import/no-extraneous-dependencies': ['error', { devDependencies: [...configHelpers.devDependenciesFiles] }],
+    'jsx-a11y/tabindex-no-positive': 'error',
+    'max-len': [
+      'error',
+      {
+        ignorePattern: [
+          'require(<.*?>)?\\(',
+          'https?:\\/\\/',
+          '^(import|export) \\{ \\w+( as \\w+)? \\} from',
+          '^import \\* as',
+          '^\\s+(<path )?d=',
+          '!raw-loader!',
+          '\\bdata:image/',
+        ].join('|'),
+        code: 120,
+      },
+    ],
+    'no-alert': 'error',
+    'no-bitwise': 'error',
+    'no-caller': 'error',
+    'no-console': 'error',
+    'no-constant-condition': 'error',
+    'no-debugger': 'error',
+    'no-duplicate-case': 'error',
+    'no-empty': 'error',
+    'no-eval': 'error',
+    'no-new-wrappers': 'error',
+    'no-restricted-properties': [
+      'error',
+      { object: 'describe', property: 'only', message: 'describe.only should only be used during test development' },
+      { object: 'it', property: 'only', message: 'it.only should only be used during test development' },
+    ],
+    'no-shadow': ['error', { hoist: 'all' }],
+    'no-var': 'error',
+    'prefer-const': 'error',
+    'prefer-arrow-callback': 'error', // tslint: no-function-expression
+    radix: ['error', 'always'],
+    // Use className and provide css rules instead of using inline styles.
+    // TODO: this doesn't forbid style on built-in elements--do we care?
+    'react/forbid-component-props': ['error', { forbid: ['style'] }], // tslint: jsx-ban-props
+    'react/jsx-key': 'error',
+    'react/jsx-no-bind': [
+      'error',
+      {
+        allowArrowFunctions: false, // tslint: jsx-no-lambda
+        allowFunctions: false,
+        allowBind: false,
+      },
+    ],
+    'react/no-string-refs': 'error',
+    'react/self-closing-comp': 'error',
+
+    // TODO: find equivalents to these tslint rules
+    // used in certain packages:
+    // "import-blacklist": [true, { "../../Styling": ["FontSizes"] }]
+
+    // airbnb or other config overrides (some temporary)
+    // TODO: determine which rules we want to enable, and make needed changes (separate PR)
+    'class-methods-use-this': 'off',
```

Comment:
A dev won't know what each of these do. It would be helpful to add docs on these. Separate PR is fine.

Reply:
I don't think we need to add docs on every disabled rule right now. Eventually the goal would be to go through the rules and either enable them, or permanently disable, and at that point probably add a comment explaining why.